docs/source/tutorials_source/package/tutorial_custom_augmentations.py

```diff
@@ -317,10 +359,28 @@ def tiff_loader(f):
     num_workers=num_workers
 )
 
-# get vector representations for all images in the dataset
-encoder.to('cuda')
-embeddings, _, fnames = encoder.embed(dataloader_test, device='cuda')
-embeddings = normalize(embeddings)
+# next we add a small helper function to generate embeddings of our images
+def generate_embeddings(model, dataloader):
```

Comment:
Just a brainfart: If we use `generate_embeddings()` in every tutorial, why not have it somewhere as a common function?

Furthermore, I would then change the input argument `model` to `backbone`, to not rely on the argument `model` having the attribute `backbone`.

Reply:
> Just a brainfart: If we use generate_embeddings() in every tutorial, why not have it somewhere as a common function?

Depends a bit on the goal of the tutorials. I think it is ok to show the user how to generate embeddings in detail. Especially now that we switched to low-level building blocks. But I agree that it generates a lot of copy paste code.

> I would then change the input argument model to backbone, to not rely on the argument model having the attribute backbone.

I agree that it would be nicer to pass only the backbone but it would add more complexity elsewhere. The update code would be:

```
model.eval()
backbone = model.backbone
device = model.device # backbone has no device attribute
generate_embeddings(backbone, dataloader, device)
```

The issue is that backbone is juste a `nn.Sequence` and has not the usual attributes like `.eval()` and `.device`. This would add even more copy paste code to the tutorials.

Reply:
Good point, the model should be kept.